src/executor/executor.py

```diff
@@ -224,81 +232,85 @@ def _start_dem_units(self,timestep=0,types=0,restart=False):
+        if restart:
+            cud.input_staging  = [{'source': 'pilot:///in.restart_from_%d'%timestep,
+                                   'target':'unit:///in.restart_from_%d'%timestep,
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/shell_closed.stl',
+                                   'target':'unit:///shell_closed.stl',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/shell',
+                                   'target':'unit:///shell',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/impeller',
+                                   'target':'unit:///impeller',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/impeller_coarse.stl',
+                                   'target':'unit:///impeller_coarse.stl',
+                                   'action'  :rp.LINK},
+                                ]
+            cud.cores          = self._DEMcores
+            cud.mpi            = True
+        else:
+            cud.input_staging  = [{'source': self._pathtoLIGGHTSinputs+'/in.2_sim_new',
+                                   'target':'unit:///in.2_sim_new',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/shell_closed.stl',
+                                   'target':'unit:///shell_closed.stl',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/shell',
+                                   'target':'unit:///shell',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/impeller',
+                                   'target':'unit:///impeller',
+                                   'action'  :rp.LINK},
+                                  {'source': self._pathtoLIGGHTSinputs+'/impeller_coarse.stl',
+                                   'target':'unit:///impeller_coarse.stl',
+                                   'action'  :rp.LINK},
+                                ]
+            cud.cores          = self._DEMcores
+            cud.mpi            = True
+        self._logger.debug('DEM unit description: %s'%cud.as_dict())
+        # Submit the first unit and wait until it staged its input files. Waiting is
+        # needed so that we can get the path of the unit and pass it to the DEM monitor.
+        dem_unit = self._umgr.submit_units(cud)
+        # This line blocks the execution until the DEM unit has a path.
+        self._logger.info('Waiting DEMs to start running')
+        self._umgr.wait_units(uids=[dem_unit.uid],state=[rp.AGENT_SCHEDULING_PENDING])
```

Comment:
Question: The info msg says 'waiting to start running'. How do you define running here? My suggestion would be to use 'executing' or 'scheduling' since they will meaningful with the state that you wait for in wait_units()

Reply:
Okay!